packages/server-wallet/src/db/migrations/20201014174039_objectives.ts

```diff
@@ -0,0 +1,20 @@
+import * as Knex from 'knex';
+
+const tableName = 'objectives';
+
+export async function up(knex: Knex): Promise<any> {
+  return knex.schema.createTable(tableName, function(table) {
+    table.string('objective_id').primary();
+    table.string('status');
+    table.string('type');
+    table.string('target_channel_id');
```

Comment:
Why is this not a foreign key?

Reply:
See #2708 . I guess it's not 100% clear to me that we want to enforce that constraint. I might want to store an objective that refers to a channel that I do not yet know about?

Reply:
Foreign key constraints are about _referential integrity_. If this objective refers to a channel, the constraint enforces that the channel exists.

If we want to enforce that, this should be a foreign key.

Reply:
Correct. I'm saying that I'm not sure we want to enforce that.

Anyways, we _will_ enforce that for the time being (current tests pass). We can revisit if needs be.